### DISCUSSION TITLE
is there any way to attach a lsp to current buffer without checking if cwd checking fails

### DISCUSSION THREAD

**Discussion by muneebusmani** [neovim/neovim#29152] [Q&A]
is there any way to attach a lsp to current buffer without checking  project root , because sometimes i do core php scripts and lsp doesnt get attached if i dont initialze a git repo or composer.json

**Comment by adigitoleo** [comment]
I assume you use nvim-lspconfig and phpactor. From `:h lspconfig-setup`:

> The `setup{}` function takes a table which contains a superset of the keys
> listed in `:help vim.lsp.start_client()` with the following unique entries:
> 
> - {root_dir}
> 
>     `function(filename, bufnr)`
> 
>     Returns either a filepath (string) or nil. The language server will only
>     start if the function returns a filepath.
> 
>     If a root directory (string) is returned which is unique from any
>     previously returned root_dir, a new server will be spawned with that
>     root directory. See |lspconfig-root-detection| for more details

So to answer the question: no it doesn't seem possible to start the lsp without setting a file path as the project root. However, you might have some luck playing with the `root_dir` function. See also `:h lspconfig-single-file-support`, which indicates that an empty `root_dir` may be supported by some lsp's (I don't think phpactor is one of them).

**Reply by muneebusmani** [reply]
i tried asking this question in phpactor github and those guys seemed clueless and said this issue is from nvim-lspconfig, not from our side, thats why i asked it here
